Automattic/wp-calypso
# Domain picker: cleanup and analytics fix

#### Changes proposed in this Pull Request
* Cleanup unused props in FSE.
* Set `ui_algo: "/gutenboarding/editor_domain_modal"` prop for `calypso_traintracks_render` when rendering domain suggestions in FSE.
* Add a selector on `domain-suggestions` store to get `vendor` value provided by wrapper UI at registration.
* Remove `vendor` prop from `DomainPicker`.

#### Testing instructions
* Follow testing instructions in https://github.com/Automattic/wp-calypso/pull/42951 to enable domain picker in editor.
* Domain Picker inside the editor and Gutenboarding should work as before.
* Debug analytics within the editor and check if `calypso_traintracks_render` event is sending `ui_algo` correctly.
